src/v/kafka/server/quota_manager.cc

```diff
@@ -35,6 +40,109 @@ using namespace std::chrono_literals;
 
 namespace kafka {
 
+template<typename clock>
+class client_quotas_probe {
+public:
+    client_quotas_probe() = default;
+    client_quotas_probe(const client_quotas_probe&) = delete;
+    client_quotas_probe& operator=(const client_quotas_probe&) = delete;
+    client_quotas_probe(client_quotas_probe&&) = delete;
+    client_quotas_probe& operator=(client_quotas_probe&&) = delete;
+    ~client_quotas_probe() noexcept = default;
+
+    void setup_metrics() {
+        namespace sm = ss::metrics;
+
+        if (config::shard_local_cfg().disable_metrics()) {
```

Comment:
Does this work with `all_metrics_groups` and `config::shard_local_cfg().disable_public_metrics()`?